types/options.go

```diff
@@ -418,8 +418,11 @@ func ReloadConfigurationFile(configFile string, storeOptions *StoreOptions) erro
 	meta, err := toml.DecodeFile(configFile, &config)
 	if err == nil {
 		keys := meta.Undecoded()
-		if len(keys) > 0 {
-			logrus.Warningf("Failed to decode the keys %q from %q", keys, configFile)
+		for _, key := range keys {
+			if key.String() == "storage.options.thinpool" {
+				continue
+			}
+			logrus.Warningf("Failed to decode the keys %q from %q", key, configFile)
```

Comment:
This seems to be a change in behavior, previously it would have logged all errors in one line now it will log each key individually. This is fine for me but you should `/s/keys/key` in the message then.